### PR TITLE
feat: api client in react

### PR DESCRIPTION
**Pull request by amritk**
Added a react client-only wrapper for the api-client.

@cameronrohani I just copied over the css file you had in the other PR, needs some 🪄 

Still needs:

- [ ] style (@cameronrohani)
- [x] documentation
- [x] typescript

### CODE REVIEW

File: packages/api-client-react/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Scalar API Client
+
```
Comment (amritk) [inline packages/api-client-react/README.md:2]: will whip this up

**Comment by hanspagel** [comment]
Do you need help with the documentation? If so, just ping me. ✌️

File: packages/api-client-react/README.md (hunk 1 above)
Comment (amritk) [reply packages/api-client-react/README.md:2]: @hanspagel hows this for docs?

File: packages/api-reference/src/hooks/useSpec.test.ts [hunk 2]
```diff
@@ -101,6 +101,7 @@ describe('useSpec', () => {
 
     await new Promise((resolve) => {
       watch(rawSpecRef, (value) => {
+        if (!value) return
```
Comment (amritk) [inline packages/api-reference/src/hooks/useSpec.test.ts:104]: @hanspagel there was no value in here which was causing an error. Is this value allowed to be falsey?

**Pull request closed without merging**